app/jenkins-for-jira-ui/src/components/ConnectionPanel/ConnectionPanelTop.tsx

```diff
@@ -24,39 +27,97 @@ const connectedStateColors: Record<ConnectedState, { textColor: string; backgrou
 	[ConnectedState.PENDING]: { textColor: '#a54900', backgroundColor: '#fff7d6' }
 };
 
-const ConnectionPanelTop = ({ connectedState, ipAddress, name }: ConnectionPanelTopProps): JSX.Element => {
+const ConnectionPanelTop = ({
+	server,
+	refreshServers
+}: ConnectionPanelTopProps): JSX.Element => {
+	const connectedState = server.connectedState || ConnectedState.PENDING;
 	const { textColor, backgroundColor } = connectedStateColors[connectedState];
+	const [serverToDisconnect, setServerToDisconnect] = useState<JenkinsServer>();
+	const [showConfirmServerDisconnect, setShowConfirmServerDisconnect] = useState(false);
+	const [isLoading, setIsLoading] = useState(false);
+
+	const onClickDisconnect = async (serverToDelete: JenkinsServer) => {
+		setServerToDisconnect(serverToDelete);
+		setShowConfirmServerDisconnect(true);
+	};
+
+	const disconnectJenkinsServerHandler = async (
+		serverToDelete: JenkinsServer
+	) => {
+		setIsLoading(true);
+
+		try {
+			await disconnectJenkinsServer(serverToDelete.uuid);
+		} catch (e) {
+			console.log('Failed to disconnect server', e);
```

Comment:
does setIsLoading(false); need to fire here or its handled other ways?

Reply:
Ooooh good question. This is the same functionality we currently use in the existing flow **but** I don't believe we have an error state if the request fails so from a quick look at what we have, I guess users would just see a never-ending loader. 🤔 My guess is I'll have to handle it better here, and error states are something I want to discuss with Sarah (so far we only have the happiest of happy paths) buuuuut #tomorrowsproblem